agent/agent.go

```diff
@@ -482,3 +481,46 @@ func (a *Agent) healthzHandler(w http.ResponseWriter, r *http.Request) {
 		w.WriteHeader(http.StatusServiceUnavailable)
 	}
 }
+
+func (a *Agent) populateSourceCache(ctx context.Context) {
```

Comment:
Should we return the errors that are generated by this function, rather than logging and burying them?

Just by looking at the nature of the errors, they seem fairly fatal.